tools/slogloggercheck/main.go

```diff
@@ -110,38 +113,187 @@ type visitor struct {
+// and String() method calls in logging functions.
 func (v *visitor) Visit(node ast.Node) ast.Visitor {
 	if node == nil {
 		return nil
 	}
 
-	// Look for selector expressions (package.identifier)
-	selectorExpr, ok := node.(*ast.SelectorExpr)
-	if !ok {
-		return v
+	// Check for DefaultSlogLogger usage
+	if selectorExpr, ok := node.(*ast.SelectorExpr); ok {
+		// Check if it's a reference to logging.DefaultSlogLogger
+		if ident, ok := selectorExpr.X.(*ast.Ident); ok {
+			if ident.Name == "logging" && selectorExpr.Sel.Name == "DefaultSlogLogger" {
+				// Check if there's a justification comment nearby
+				pos := v.fileSet.Position(node.Pos())
+				if !hasJustificationComment(v.fileSet, node, requiredComment) {
+					fmt.Printf("%s:%d: direct use of logging.DefaultSlogLogger without justification comment\n",
+						v.fileName, pos.Line)
+					fmt.Printf("  Add a comment with '%s <reason>' to justify this usage\n", requiredComment)
+					v.found = true
+				}
+			}
+		}
 	}
 
-	// Check if it's a reference to logging.DefaultSlogLogger
-	if ident, ok := selectorExpr.X.(*ast.Ident); ok {
-		if ident.Name == "logging" && selectorExpr.Sel.Name == "DefaultSlogLogger" {
-			// Check if there's a justification comment nearby
-			pos := v.fileSet.Position(node.Pos())
-			if !hasJustificationComment(v.fileSet, node) {
-				fmt.Printf("%s:%d: direct use of logging.DefaultSlogLogger without justification comment\n",
-					v.fileName, pos.Line)
-				fmt.Printf("  Add a comment with '%s <reason>' to justify this usage\n", requiredComment)
-				v.found = true
+	// Check for String() calls in logging functions
+	if callExpr, ok := node.(*ast.CallExpr); ok {
+		// Check if it's a logger method call (e.g., logger.Debug, logger.Info)
+		if sel, ok := callExpr.Fun.(*ast.SelectorExpr); ok {
+			// Check if the method is a logging method (Debug, Info, Warn, Error)
+			if isLoggingMethod(sel.Sel.Name) {
+				// Examine arguments for String() calls
+				v.checkLoggingCallArgs(sel.Sel.Name, callExpr.Args)
 			}
 		}
 	}
 
 	return v
 }
 
+// checkLoggingCallArgs examines the arguments of a logging call for String() method invocations
+func (v *visitor) checkLoggingCallArgs(methodName string, args []ast.Expr) {
+	for i := 0; i < len(args); i++ {
+		arg := args[i]
+
+		// Look for direct String() calls in the argument
+		if stringCall, isPtr := v.findStringMethodCall(arg); stringCall != nil && !isPtr {
+			pos := v.fileSet.Position(stringCall.Pos())
+			if !hasJustificationComment(v.fileSet, arg, stringMethodComment) {
+				fmt.Printf("%s:%d: String() method call on a non-pointer in %s() logging function MUST be avoided\n",
+					v.fileName, pos.Line, methodName)
+				fmt.Printf("  This will cause unnecessary string conversion even when the log level is disabled\n")
+				fmt.Printf("  Simply don't call the String()\n")
+				fmt.Printf("  If not possible you can justify with '%s <reason>' to justify this usage\n", stringMethodComment)
```

Comment:
I'd suggest instructing the developer the exact form to use if justification is necessary. Otherwise for arbitrary contributors it'll be hard for them to understand what it means to "justify the usage".